src/components/searchBar.tsx

```diff
@@ -1,65 +1,57 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './searchBar.css';
 
 const SEARCH_BAR_HEIGHT = '36px';
 const MARGIN_TOP_BOTTOM = '10px';
-export const SEARCH_HEIGHT = `${SEARCH_BAR_HEIGHT} + 2 * ${MARGIN_TOP_BOTTOM}`;
 
 interface ISearchBarProps {
   placeholder: string;
   searchHandler: (search: string) => void;
 }
 
-interface ISearchBarState {
-  value: string;
-}
-
-class SearchBar extends React.Component<ISearchBarProps, ISearchBarState> {
-  constructor(props: ISearchBarProps) {
-    super(props);
-    const savedValue = localStorage.getItem('searchBarValue');
-    this.state = { value: savedValue || '' };
-  }
+function SearchBar({ placeholder, searchHandler }: ISearchBarProps) {
+  const [value, setValue] = useState('');
 
-  handleChange: React.ChangeEventHandler<HTMLInputElement> = (event) => {
-    this.setState({ value: event.target.value });
+  const handleChange: React.ChangeEventHandler<HTMLInputElement> = (event) => {
+    setValue(event.target.value);
   };
 
-  handleClearClick: React.MouseEventHandler<HTMLButtonElement> = (event) => {
+  const handleClearClick: React.MouseEventHandler<HTMLButtonElement> = (event) => {
     event.preventDefault();
-    this.setState({ value: '' });
+    setValue('');
   };
 
-  componentWillUnmount = () => {
-    localStorage.setItem('searchBarValue', this.state.value);
-  };
-
-  handleSearchSubmit: React.FormEventHandler<HTMLFormElement> = (event) => {
+  const handleSearchSubmit: React.FormEventHandler<HTMLFormElement> = (event) => {
     event.preventDefault();
-    this.props.searchHandler(this.state.value);
+    searchHandler(value);
   };
 
-  render() {
-    return (
-      <form
-        className="search-container"
-        style={{ height: SEARCH_BAR_HEIGHT, margin: `${MARGIN_TOP_BOTTOM} auto` }}
-        data-testid="search-bar"
-        onSubmit={this.handleSearchSubmit}
-      >
-        <button type="submit" className="button search-button"></button>
-        <input
-          className="search"
-          type="text"
-          value={this.state.value}
-          placeholder={this.props.placeholder}
-          autoComplete="off"
-          onChange={this.handleChange}
-        />
-        <button className="button clear-button" onClick={this.handleClearClick}></button>
-      </form>
-    );
-  }
+  useEffect(() => {
+    const savedValue = localStorage.getItem('searchBarValue');
+    setValue(savedValue || '');
+  }, []);
+
+  useEffect(() => localStorage.setItem('searchBarValue', value));
```

Comment:
Make sense to add dependency array. It will save `searchBarValue` only when value has changed and not on every re-render. 

`useEffect(() => localStorage.setItem('searchBarValue', value), [value]);`

I recommend to use this eslint plugin https://reactjs.org/docs/hooks-rules.html#eslint-plugin